parsers/rhea/rhea.go

```diff
@@ -0,0 +1,430 @@
+				newCompound.Position = description.Position
+				newCompound.Name = description.Name
+				newCompound.HTMLName = description.HTMLName
+				newCompound.Formula = description.Formula
+				newCompound.Charge = description.Charge
+				newCompound.ChEBI = description.ChEBI.Resource
+				rhea.Compounds = append(rhea.Compounds, newCompound)
+			}
+		}
+		for _, containsx := range description.ContainsX {
+			if !strings.Contains(containsx.XMLName.Local, "contains") {
+				continue
+			}
+			newReactionParticipant, err := NewReactionParticipant(description, containsx, compoundParticipantMap)
+			if err != nil {
+				return Rhea{}, err
+			}
+			rhea.ReactionParticipants = append(rhea.ReactionParticipants, newReactionParticipant)
+		}
+	}
+	return rhea, nil
+}
+
+// ReadRhea reads in a a gzip'd Rhea dump (https://www.rhea-db.org/help/download) into bytes.
+func ReadGzippedXml(gzipPath string) ([]byte, error) {
+	// Get gz'd file bytes
+	xmlFile, err := os.Open(gzipPath)
+	if err != nil {
+		return []byte{}, err
+	}
+
+	// Decompress gz'd file
+	r, err := gzip.NewReader(xmlFile)
+	if err != nil {
+		return []byte{}, err
+	}
+
+	// Read decompressed gz'd file
+	rheaBytes, err := ioutil.ReadAll(r)
+	if err != nil {
+		return []byte{}, err
+	}
+	return rheaBytes, nil
+}
+
+/******************************************************************************
+
+JSON functions
+
+These functions simply export the Rhea database as a JSON file for consumption
+in different programs.
+
+******************************************************************************/
+
+// Export exports Rhea as a JSON file
+func (rhea *Rhea) Export() ([]byte, error) {
+	rheaJSON, err := json.Marshal(rhea)
+	if err != nil {
+		return []byte{}, err
+	}
+	return rheaJSON, nil
+}
+
+/******************************************************************************
+
+Rhea2Uniprot tsv
+
+Rhea conveniently provides a TSV list of reaction IDs to Uniprot accession numbers.
+These can be used to map Rhea reactions to protein sequences, which is very useful
+for actually building synthetic circuits. While these parsers are very basic TSV
+parsing, they are convenient to use with the rhea2uniprot data dumps specifically.
+
+The sprot and trembl data dumps are available at:
+https://ftp.expasy.org/databases/rhea/tsv/rhea2uniprot_sprot.tsv
+https://ftp.expasy.org/databases/rhea/tsv/rhea2uniprot_trembl.tsv.gz
+
+The decompressed gzip side of the rhea2uniprot_trembl.tsv.gz file, as of Apr 30, 2021
+is 678M. Because it is fairly large, the parser is implemented to stream in the file,
+passing the parsed lines into a channel, to save on memory.
+
+The TSV is structured like:
+
+`
+RHEA_ID DIRECTION       MASTER_ID       ID
+10008   UN      10008   O17433
+10008   UN      10008   O34564
+...
+`
+
+******************************************************************************/
+
+// Rhea2Uniprot represents a single line of the TSV file.
+type Rhea2Uniprot struct {
```

Comment:
This belongs in a different PR.

PR's should be compact. Easier to review, Easier to revert in case it introduces a bug.

When a PR is landed it becomes a sort of documentation for future engineers when they search by `blame` so we want to minimize the cognitive load of future maintainers. 

If you are blocked, you can look into stacked PR's, but they take a bit of `rebase` kung fu to get used to: https://blog.logrocket.com/using-stacked-pull-requests-in-github/

Reply:
Fine to leave it in for now, but should use stacked PR's in the future